index.bs

```diff
@@ -555,6 +703,234 @@ Report Long Animation Frames {#loaf-processing-model}
+</div>
+
+<div algorithm="Set source URL for script block">
+    To <dfn>set source url for script block</dfn> given a [=script timing info=] |scriptTimingInfo|, a [=/script=] |script|, and a [=/URL=] |url|:
+        1. If |url|'s [=url/scheme=] is "`http`" or "`https`", then set |scriptTimingInfo|'s [=script timing info/source url=] to |script|'s [=script/base URL=].
+        1. Otherwise, if |url|'s [=url/scheme=] is "`blob`" or "`data`" then set |scriptTimingInfo|'s [=script timing info/source url=] to the [=concatenate|concatenation=] of  « |url|'s [=url/scheme=], ":"" ».
+</div>
+
+<div algorithm="Report classic script execution">
+    To <dfn>report classic script creation</dfn> given a [=/script=] |script| and a [=/URL=] |originalSourceURL|:
+        1. [=Create script entry point=] with |script|'s [=script/settings object=], "`classic-script`",
+            and the following step given a [=script timing info=] |scriptTimingInfo|:
+            [=Set source url for script block=] given |scriptTimingInfo|, |script|, and |originalSourceURL|.
+</div>
+
+<div algorithm="Report classic script execution start">
+    To <dfn>report classic script execution start</dfn> given a [=classic script=] |script|:
+        1. Let |settings| be |script|'s [=script/settings object=].
+        1. If |script|'s [=classic script/muted errors=] is true, then return.
+        1. If |settings| is not a {{Window}}, then return.
+        1. Let |document| be |settings|'s {{Window/document}}.
+        1. Let |frameTimingInfo| be |document|'s [=relevant frame timing info=].
+        1. If |frameTimingInfo| is null or if |frameTimingInfo|'s [=frame timing info/pending script=] is not null, then return.
+        1. Assert: |frameTimingInfo|'s [=frame timing info/pending script=]'s [=script timing info/type=] is "`classic-script`".
+        1. Set |frameTimingInfo|'s [=frame timing info/pending script=]'s [=script timing info/execution start time=] to the [=unsafe shared current time=].
+</div>
+
+<div algorithm="Report module script execution start">
+    To <dfn>report module script execution start</dfn> given a [=module script=] |script|:
+        [=Create script entry point=] with |script|'s [=script/settings object=], "`module-script`",
+            and the following step given a [=script timing info=] |scriptTimingInfo|:
+            1. Set |scriptTimingInfo|'s [=script timing info/execution start time=] to |script|'s |scriptTimingInfo|'s [=script timing info/start time=].
+            1. [=Set source url for script block=] given |scriptTimingInfo|, |script|, and |script|'s [=script/base URL=].
+</div>
+
+<div algorithm="Create script entry point">
+    To <dfn>create script entry point</dfn> given an [=environment settings object=] |settings|,
+        a {{ScriptTimingType}} |type|, and |steps|,
+        which is an algorithm that takes a [=script timing info=] and an optional [=frame timing info=]:
```

Comment:
It took me a while to understand the way that "optional" is working here -- since `frameTimingInfo` is unconditionally passed to the callback, there's no reason for the callback to ever check that it exists, or is non-null. (So, not optional on either side of that function call).

It's more like the JS sense, where all arguments are "optional"; you don't have to name positional parameters that you aren't interested in, and the arguments on each side of the function call don't have to match at all.